src/views/CustomToken.vue

```diff
@@ -164,25 +164,25 @@ export default {
 
       if (this.existingAsset) {
         customToken = this.existingAsset
-      } else if (this.activeNetwork === 'mainnet') {
+      } else if (this.activeNetwork === 'mainnet' && this.contractAddress) {
         try {
-          const result = await axios.get(`https://api.ethplorer.io/getTokenInfo/${this.contractAddress}?apiKey=freekey`)
-          const { symbol, name, decimals } = result.data
-          customToken = { symbol, name, decimals: parseInt(decimals), chain: 'ethereum' }
+          const { symbol, name, decimals } = await TOKEN_DETAILS[this.chain].getDetails(this.contractAddress)
```

Comment:
`TOKEN_DETAILS` makes it feel like this is some sort of constant value instead of a module 
i think it should be more like `tokenDetailProviders` or something like that